docs/content/en/migrating.md

```diff
@@ -17,6 +17,8 @@ Follow this guide to upgrade from one major version to the other.
 
 ### `seo` option has been removed. Use `$nuxtI18nHead({ addDirAttribute: true, addSeoAttributes: true })` instead.
 
+### `onlyOnRoot` and `onlyOnNoPrefix` has been removed. It's features are combined into `redirectOn` instead. Use `redirectOn: 'root|no_prefix'` instead.
```

Comment:
I think I would prefer the default value to be `'all'` than `null`.

Also would use `no prefix` instead of `no_prefix`. I guess you wanted to match the value for the `no_prefix` strategy but I'm not sure that's a good idea. Can be potentially more confusing than anything.